core/workspace.js

```diff
@@ -131,11 +131,16 @@ Blockly.Workspace = function(opt_options) {
    */
   this.marker_ = new Blockly.MarkerCursor();
 
-  // Set the default theme. This is for headless workspaces. This will get
-  // overwritten by the theme passed into the inject call for rendered workspaces.
-  if (!Blockly.getTheme()) {
-    Blockly.setTheme(Blockly.Themes.Classic);
-  }
+  /**
+   * Object in charge of storing and updating the workspace theme.
+   * @type {!Blockly.ThemeManager}
+   * @protected
+   */
+  this.themeManager_ = this.options.parentWorkspace ?
+      this.options.parentWorkspace.getThemeManager() :
+      new Blockly.ThemeManager(this.options.theme);
```

Comment:
Should we set a default theme here if there are none are on the options? There was a bug that the code we deleted fixed for headless workspaces. Issue #2605

Reply:
Yep, agreed.